src/status_im2/contexts/syncing/setup_syncing/view.cljs

```diff
@@ -38,31 +38,37 @@
-                        (when (sync-utils/valid-connection-string? connection-string)
-                          (reset! timestamp (* 1000 (js/Math.ceil (/ (datetime/timestamp) 1000))))
-                          (reset! delay 1000)
-                          (reset! code connection-string)))
-        clock         (fn []
-                        (if (pos? (- code-valid-for-ms
-                                     (- (* 1000 (js/Math.ceil (/ (datetime/timestamp) 1000)))
-                                        @timestamp)))
-                          (swap! valid-for-ms (fn [_]
-                                                (- code-valid-for-ms
-                                                   (- (* 1000
-                                                         (js/Math.ceil (/ (datetime/timestamp) 1000)))
-                                                      @timestamp))))
-                          (reset! delay nil)))
-        cleanup-clock (fn []
-                        (reset! code nil)
-                        (reset! timestamp nil)
-                        (reset! valid-for-ms code-valid-for-ms))]
-
+  (let [{:keys [customization-color]} (rf/sub [:profile/multiaccount])
+        valid-for-ms                  (reagent/atom code-valid-for-ms)
+        code                          (reagent/atom nil)
+        delay                         (reagent/atom nil)
+        timestamp                     (reagent/atom nil)
+        set-code                      (fn [connection-string]
+                                        (when (sync-utils/valid-connection-string? connection-string)
+                                          (reset! timestamp (* 1000
+                                                               (js/Math.ceil (/ (datetime/timestamp)
+                                                                                1000))))
+                                          (reset! delay 1000)
+                                          (reset! code connection-string)))
+        clock                         (fn []
+                                        (if (pos? (- code-valid-for-ms
+                                                     (- (* 1000
+                                                           (js/Math.ceil (/ (datetime/timestamp) 1000)))
+                                                        @timestamp)))
+                                          (swap! valid-for-ms (fn [_]
+                                                                (- code-valid-for-ms
+                                                                   (- (* 1000
+                                                                         (js/Math.ceil
+                                                                          (/ (datetime/timestamp) 1000)))
+                                                                      @timestamp))))
+                                          (reset! delay nil)))
```

Comment:
This part is hard to read. Maybe extract to utils file and break it up into smaller functions if possible

Reply:
sure, this was previously here so I will create a follow up to address this. Also it seems this code will be better as being dispatched as it will break the need for so many nested callbacks